### PR TITLE
Endpoint to expose interfaces' statistics

### LINKED ISSUE #343: Endpoint to provide container's interfaces statistics
Allow libnetwork client to access container side interface statistics

### PR DESCRIPTION
**Pull request by aboch**
Closes #343 

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

File: endpoint.go [hunk 1]
```diff
@@ -51,6 +51,9 @@ type Endpoint interface {
 
 	// Delete and detaches this endpoint from the network.
 	Delete() error
+
+	// Retrieve the interfaces' statistics from the sandbox
+	Statistics() (map[string]*sandbox.InterfaceStatistics, error)
```
Comment (aboch) [inline endpoint.go]: Will move this under InterfaceInfo.

Comment (mrjana) [inline endpoint.go]: Why?

**Comment by mrjana** [comment]
LGTM

File: sandbox/interface_linux.go [hunk 2]
```diff
@@ -311,3 +340,25 @@ func setInterfaceRoutes(iface netlink.Link, i *nwIface) error {
...
+	)
+
+	regex := fmt.Sprintf(base, ifName)
+	re := regexp.MustCompile(regex)
+	line := re.FindString(data)
+
+	_, err := fmt.Sscanf(line, "%s %d %d %d %d %d %d %d %d %d %d %d %d %d %d %d %d",
+		&bktStr, &i.RxBytes, &i.RxPackets, &i.RxErrors, &i.RxDropped, &bkt, &bkt, &bkt,
+		&bkt, &i.TxBytes, &i.TxPackets, &i.TxErrors, &i.TxDropped, &bkt, &bkt, &bkt, &bkt)
+
```
Comment (mavenugo) [inline sandbox/interface_linux.go]: Not too comfortable with parsing the formatted file to get to the desired data. Seems very fragile. Libcontainer used another approach which seems more appropriate :
` // Reads the specified statistics available under /sys/class/net/<EthInterface>/statistics`

Comment (aboch) [inline sandbox/interface_linux.go]: If the sys/class/net/<dev>/<counter> file structure is there in each network namespace in the oldest kernel docker supports, I am fine switching to that.

Comment (mavenugo) [inline sandbox/interface_linux.go]: Just checked that centos 6.6 doesnt have per-netns sysctl support in its kernel version.
Hence we wont be able to use `/sys/class/net/<EthInterface>/statistics`

Comment (aboch) [inline sandbox/interface_linux.go]: Thanks for checking.

Let me add a comment so that we keep track in the code why we went for this approach.

**Comment by mavenugo** [comment]
@aboch pls remove the dnet binary :)

**Comment by mavenugo** [comment]
LGTM

**Pull request merged**